src/parser/FStar.Parser.AST.fs

```diff
@@ -670,22 +668,31 @@ and binder_to_string x =
   | TAnnotated(i,t)
   | Annotated(i,t) -> Util.format2 "%s:%s" ((string_of_id i)) (t |> term_to_string)
   | NoName t -> t |> term_to_string in
-  Util.format2 "%s%s" (aqual_to_string x.aqual) s
+  Util.format3 "%s%s%s"
+    (aqual_to_string x.aqual)
+    (if x.battributes=[] then "" else attrs_opt_to_string (Some x.battributes))
+    s
 
 and aqual_to_string = function
   | Some Equality -> "$"
   | Some Implicit -> "#"
-  | Some (Meta (Arg_qualifier_meta_tac t)) -> "#[" ^ term_to_string t ^ "]"
-  | Some (Meta (Arg_qualifier_meta_attr t)) -> "[@@" ^ term_to_string t ^ "]"
+  | Some (Meta t) -> "#[" ^ term_to_string t ^ "]"
   | None -> ""
 
+and attr_list_to_string = function
+  | [] -> ""
+  | l -> attrs_opt_to_string (Some l)
+
 and pat_to_string x = match x.pat with
-  | PatWild None -> "_"
-  | PatWild _ -> "#_"
+  | PatWild (None, attrs) -> "_" ^ (attr_list_to_string attrs)
```

Comment:
Those two lines are a little inconsistent. I think attributes should go before "_"?

Reply:
Thanks @mateuszbujalski , fixed.